src/test/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/resource/GoalControllerTest.kt

```diff
@@ -8,6 +8,7 @@ import org.mockito.kotlin.any
 import org.mockito.kotlin.given
 import org.mockito.kotlin.verify
 import uk.gov.justice.digital.hmpps.domain.aValidPrisonNumber
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.note.service.NoteService
```

Comment:
I think we can (and should?) delete this unit test altogether ...

Its the only controller that has a unit test class; everything else is tested by the integration tests.

The original thinking was that the controllers were so lightweight and slim that unit tests offered no value (especially considering the integrations which are the real tests we are interested in)

Then we added that error handling stuff, where the service returned an object, and depending on that object the controller would either return the resource, or return an error response resource.

Given that we've since removed that and reverted back to "standard" exception throwing and a controller advice, I really cant see any gain in this unit test class?

What do you think?

Reply:
It does seem a bit pointless. I'm all for removing tests for the sake of tests.